perf: improve chat switch performance

hurali97 opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

Chat switching is one of the areas that's complicated and with this PR, we aim to fix it partially. Which means we are able to improve the user experience with this PR. To summarise, on a mid-ranged device on main branch, we get around 7600 milliseconds and with these improvements, we get around 4100 milliseconds.

**Before:**

https://github.com/Expensify/App/assets/47336142/df21234b-ef8f-44d3-a25e-6b35fea1ca66

**After:**

https://github.com/Expensify/App/assets/47336142/56b964d8-2ba5-4dd5-9587-e7370f507af3

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/32793
PROPOSAL: https://github.com/Expensify/App/issues/32793



### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

## Manual Test:
- Open any chat
- The transition should feel smooth as compared to main
- Press back button
- The transition should feel smooth as compared to main

- [X] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/47336142/d958face-cdd6-48bd-b5c4-2d72b2883df9

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/47336142/ab200428-cca8-4499-8cb5-733bd5ceaf5c

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/47336142/f80eeec4-3373-4eb8-9787-b8df6b237838

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/47336142/fa101d09-95e4-4ea2-ba55-c15c55021f4d

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/47336142/156bc124-539a-4393-932e-1eafd61aefe4

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/47336142/fc40e371-dc17-47fc-afb3-27403cd70d1d

</details>

hurali97 committed:
perf: make LHNOptionsList lightweight

hurali97 committed:
feat: add util methods for report and reportactions

hurali97 committed:
perf: add InteractionManager to lazily call methods

hurali97 committed:
perf: avoid unnecessary updating onyx

hurali97 committed:
perf: memoize SidebarLinks

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
perf: avoid re-creating function and object instances

hurali97 committed:
perf: delay updateComment to run on mount when JS thread is idle

hurali97 committed:
fix: linting and ts issues

hurali97 committed:
perf: apply perf improvements from pr 30168

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
fix: personal details

TMisiukiewicz commented on src/components/LHNOptionsList/OptionRowLHNData.js:
```diff
@@ -127,4 +112,17 @@ OptionRowLHNData.displayName = 'OptionRowLHNData';
  * use it to prevent re-renders from parent re-renders.
  */
-export default React.memo(OptionRowLHNData);
+export default withOnyx({
+    policy: {
+        key: ({fullReport}) => `${ONYXKEYS.COLLECTION.POLICY}${fullReport.policyID}`,
+        initialValue: {},
+    },
+    comment: {
+        key: ({reportID}) => `${ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT}${reportID}`,
+        initialValue: '',
+    },
+    transaction: {
+        key: ({transactionID}) => `${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`,
+        initialValue: {},
+    },
+})(React.memo(OptionRowLHNData));
```
I just want to double-check it does not create any overhead when scrolling the list with large amount of items - we removed this HOC from this component before Flashlist migration, as it was constantly calling `Onyx.connect` while scrolling the list, making it really unperformant. The best option to test it would be account with as much reports as possible (e.g. Edu's newdotperf account with 15k records)

TMisiukiewicz commented on src/libs/ReportUtils.ts:
```diff
@@ -4284,7 +4284,12 @@ function navigateToPrivateNotes(report: Report, session: Session) {
     Navigation.navigate(ROUTES.PRIVATE_NOTES_LIST.getRoute(report.reportID));
 }
 
+function getReportByID(reportID: string) {
+    return allReports ? allReports[reportID] : {};
+}
+
```
is there a possibility of making this and `getReportActionsByReportID` functions easier to use by passing the ID without the Onyx key? IMO it's easier to use when you actually don't have to know the exact key

TMisiukiewicz commented on src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.js:
```diff
@@ -514,6 +508,27 @@ function ComposerWithSuggestions({
+    }, []);
+
+    const onChangeText = useCallback((text) => {
+        updateComment(text, true);
```
nit: did you measure the `updateComment` execution time? Not sure if the logic behind this function is complex so I'm just wondering if this is something that could be e.g. debounced

hurali97 replied:
We certainly can't debounced it as it's doing a setState which reflects the value in the input, if we debounce it we will have laggy input. There are other areas in this particular function which have debounce associated already for example, setting draft status.

hurali97 committed:
revert: bring back old improvements for LHN

hurali97 committed:
fix: personalDetails in HeaderView

hurali97 committed:
refactor: remove unused code

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
fix: linting

hurali97 committed:
fix: add reports to extraData in LHNOptionsList

adhorodyski commented on src/pages/home/ReportScreen.js:
```diff
@@ -507,14 +519,30 @@ export default compose(
     ),
-)(ReportScreen);
+)(
+    memo(
```
Any way we can avoid these checks? This rather indicates a bad design and can become really hard to maintain & debug in the future.

hurali97 replied:
Apparently, as of now there's no straightforward way to avoid `arePropsEqual` function. One way I can think of is to have this in Onyx selector and get only the properties we need from report for example. But this will not work if we have to make sure that `reportActions` are the same. We can handle that as well but at `renderItem` level but we also need `reportActions` before that, so it's rather complicated.

But I think we should do that and we can tackle this in part 2 of our effort to improve chat switch performance.

adhorodyski commented on src/pages/home/report/ReportFooter.js:
```diff
@@ -122,4 +119,19 @@ export default compose(
         },
     }),
-)(ReportFooter);
+)(
```
is it also required even with the lightweight report object?

hurali97 replied:
Yes the reason is that, the lightweight report object is created in `ReportActionsList` and not in `ReportScreen`. The reason is that in `ReportScreen` we are passing `report` object to a handful of utility functions and having the lightweight object here is rather complicated.

hurali97 commented:
@allroundexperts  Following are the files with diff that is expected to be removed after [this](https://github.com/Expensify/App/pull/30168) PR is merged, because that'll be duplicated then.

src/components/AnonymousReportFooter.tsx
src/components/AvatarWithDisplayName.tsx
src/components/MoneyReportHeader.js
src/components/MoneyRequestHeader.js
src/pages/home/HeaderView.js
src/pages/home/report/ReportActionCompose/SilentCommentUpdater.js
src/pages/home/report/ReportActionsListItemRenderer.js
src/pages/home/report/ReportFooter.js

mountiny commented:
@allroundexperts feel free to go ahead with the review, thanks!

allroundexperts commented:
> @allroundexperts Following are the files with diff that is expected to be removed after [this](https://github.com/Expensify/App/pull/30168) PR is merged, because that'll be duplicated then.
> 
> src/components/AnonymousReportFooter.tsx src/components/AvatarWithDisplayName.tsx src/components/MoneyReportHeader.js src/components/MoneyRequestHeader.js src/pages/home/HeaderView.js src/pages/home/report/ReportActionCompose/SilentCommentUpdater.js src/pages/home/report/ReportActionsListItemRenderer.js src/pages/home/report/ReportFooter.js

Got it. Thanks @hurali97!

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

allroundexperts commented on src/components/MoneyRequestHeader.js:
```diff
@@ -65,7 +62,8 @@ const defaultProps = {
 
-function MoneyRequestHeader({session, parentReport, report, parentReportAction, transaction, policy, personalDetails}) {
+function MoneyRequestHeader({session, parentReport, report, parentReportAction, transaction, policy}) {
+    const personalDetails = usePersonalDetails() || CONST.EMPTY_OBJECT;
```
Can't we move `EMPTY_OBJECT` to the `usePersonalDetails` hook itself?

hurali97 replied:
I am not sure of it because it was already introduced in the codebase and I have just copy/ pasted it from elsewhere. Apart from this, The hook is coming from `Expensify-App/src/components/OnyxProvider.tsx` so I am not sure whether there's a straight forward way to do that.

I can ask @kacper-mikolajczak if that's possible because he did investigate around this.

kacper-mikolajczak replied:
Hi @allroundexperts @hurali97, I've done that quite a while ago but it probably still holds true. 

The `usePersonalDetails` hook was a replacement for passing the `personalDetails` via props. Depending on a scenario, `personalDetails` prop sometimes had fallback in form of **defaultProps** and sometimes not.
In order to have full compat between those usages and not break anything, I decided to inline fallback to empty object depending on a use-case. 

Hope that helps 👍

allroundexperts commented:
This surely does improve the user experience. Switching the chats is much more snappier!

allroundexperts commented:
@hurali97 Can you resolve the conflicts please?

allroundexperts commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

allroundexperts commented:
**Bug:** On Android, sometimes a black screen appears without any skeleton loader when a new chat is opened.

https://github.com/Expensify/App/assets/30054992/15dc2e74-4d9d-4a96-af73-ea5f4e389dff

allroundexperts commented:
**Bug:** On mweb, often times, the chat switch animation jitters.

https://github.com/Expensify/App/assets/30054992/8caeb336-38ae-4343-ab6b-377845217c63

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 commented:
@allroundexperts conflicts resolved. 

Just to ask, does these two bugs happen on the main too? Especially the last one. Because I think I have encountered that on debug build on main too but I will verify again before proceeding with the fix.

hurali97 commented:
> **Bug:** On Android, sometimes a black screen appears without any skeleton loader when a new chat is opened.
> 
>  Screen.Recording.2023-12-14.at.8.55.27.AM.mov

@allroundexperts Let me know whether below looks fine 👍 

https://github.com/Expensify/App/assets/47336142/921c43c5-3ea8-4941-9b31-720902eba384

By the way, I have noticed that the Skeleton also doesn't appear on main but it's okay to fix that in this PR too.

hurali97 commented:
> **Bug:** On mweb, often times, the chat switch animation jitters.
> 
>  Screen.Recording.2023-12-14.at.8.51.45.AM.mov

@allroundexperts So I was able to repro and did a fix for it, let me know if it looks good 👍 

https://github.com/Expensify/App/assets/47336142/2e52cdf7-ef9e-4f3a-8edc-a3d0e31b7f8a

hurali97 committed:
fix: skeleton not showing and screen transition on mWeb

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
refactor: use didScreenTransitionEnd

hurali97 commented:
Reassure test is failing due to the fact that we have now updated the condition to show `ReportActionsSkeletonView` and `ReportActionsView` based on `didScreenTransitionEnd` which is a state value and it's toggled in the `transitionEnd` event listener of navigation.

Now, in Reassure we are just testing the mount of the `ReportScreen` so there's no actual navigation going on behind the scenes. Hence, the `transitionEnd` event doesn't fire and we always get the `didScreenTransitionEnd` as false. So reassure test fails since they can't find `report-actions-list` in the tree. 

Previously, this wasn't causing any failures due to the fact that we were not leveraging `didScreenTransitionEnd` in `ReportScreen`. This is however used in `SearchPage` but we don't have reassure tests for that, yet.

I have created a fix for this and I will address this in a separate PR to fix the reassure test there.

hurali97 committed:
fix: typecheck in AvatarWithDisplayName

allroundexperts reviewed (approved):
Looks good!

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
refactor: remove not needed personal details

The reason is that, we only use personalDetails to pass to AvatarWithDisplayName and now this component already have Onyx.connect to provide personalDetails directly

allroundexperts commented:
@mountiny What are we waiting for here?

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
test: fix onEntryTransition not being called in UnitTests

mountiny commented:
@allroundexperts So I was hoping to get this PR merged first https://github.com/Expensify/App/pull/30168 because here we are building atop, but its been taking really long time to conclude. @aimane-chnaif whats blocking you from completing the checklist and approving? You might be able to resolve the conflicts locally yourself

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
refactor: remove dead code

mountiny commented:
@hurali97 more conflicts

@allroundexperts would you be able to prioritize this over other issues once resolved? thanks!

allroundexperts commented:
@mountiny I'm reviewing this now.

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

mountiny commented:
@hurali97 is the performance test failing expected?

hurali97 commented:
@mountiny I can verify this on Thursday, as I have a day off tomorrow 👍  If performance test failure is not expected then I will try to fix it.

allroundexperts commented:
@hurali97 Do you know if the following is expected? Notice that the page always loads first even if it was just opened previously.

https://github.com/Expensify/App/assets/30054992/5e138077-543d-4ac9-888f-3706c26d0a33

mountiny commented:
That would be a regression from expected behaviour for sure @allroundexperts

hurali97 commented:
@allroundexperts I am working to fix the regression and also verify the reassure test. I hope to finalise this today, as I am on vacations from next week. Will keep you posted 👍

hurali97 committed:
Merge branch 'main' of https://github.com/callstack-internal/Expensify-App into hur/perf/chat-switch

hurali97 committed:
fix: skeleton loader being shown on chat switch each time

hurali97 committed:
fix: prettier

hurali97 commented:
@allroundexperts All good now 👍

allroundexperts commented:
Conflicts again 😢

jbroma committed:
Merge branch 'main' into hur/perf/chat-switch

jbroma commented:
I have read the CLA Document and I hereby sign the CLA

jbroma commented:
@allroundexperts resolved conflicts 👍

jbroma committed:
Merge branch 'main' into hur/perf/chat-switch

mountiny commented:
@allroundexperts tested again here https://expensify.slack.com/archives/C02NK2DQWUX/p1709827885676179?thread_ts=1709809294.950899&cid=C02NK2DQWUX

mountiny commented on src/pages/home/ReportScreen.js:
```diff
@@ -584,9 +599,7 @@ function ReportScreen({
                                         isEmptyChat={isEmptyChat}
                                         lastReportAction={lastReportAction}
                                     />
-                                ) : (
-                                    <ReportFooter isReportReadyForDisplay={false} />
-                                )}
+                                ) : null}
```
What will this change accomplish? Will this mean that the Composer is not going to be visible when the report screen is loading?

mountiny reviewed (approved):
Thanks @hurali97 @jbroma @allroundexperts lets be on a look out for regressions over the weekend, I feel like there could be some things that could have been missed during testing

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

mountiny commented:
@hurali97 @allroundexperts we had to revert the PR unfortunately, we can collect the issues reported and address them in the v2

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/8253954132) to production by https://github.com/luacmartins in version: 1.4.50-5 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/8253965351) to staging by https://github.com/mountiny in version: 1.4.51-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/8270531420) to production by https://github.com/luacmartins in version: 1.4.51-3 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅
